src/pet.js

```diff
@@ -0,0 +1,71 @@
+const MAX_FITNESS_LEVEL = 10;
+const MIN_HUNGER_LEVEL = 0;
+function Pet(name) {
+    this.name = name;
+    this.age = 0;
+    this.hunger = 0;
+    this.fitness = 10;
+    this.children=[];
+}
+
+Pet.prototype = {
+    get isAlive() {
+        return this.age < 30 && this.hunger < 10 && this.fitness > 0;
+    }
+}
+
+Pet.prototype.growUp = function () {
+    if(!this.isAlive){
+        throw new Error('Your pet is no longer alive :(');
+    }
+    this.age += 1;
+    this.hunger += 5;
+    this.fitness -= 3;
+}
+
+Pet.prototype.walk = function () {
+    if(!this.isAlive){
+        throw new Error('Your pet is no longer alive :(');
+    }
+    this.fitness += 4;
+    if(this.fitness === 9 || this.fitness > MAX_FITNESS_LEVEL){
+        this.fitness = MAX_FITNESS_LEVEL;
+    }
```

Comment:
This could be refactored a bit. In essence you are adding a value to the existing fitness, without allowing it go over a set limit. You can do this with Math.min, which will return the lowest of the values passed to it:
```
this.fitness = Math.min(this.fitness + 4, MAX_FITNESS_LEVEL)
```